### PR TITLE
net-p2p/biglybt: update to 3.6.0.0, add icon, and more changes

### PR DESCRIPTION
**Pull request by gentoo-setan**
---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-p2p/biglybt/biglybt-3.6.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	java-pkg-simple_src_compile
+
+	if use doc; then
+		JAVADOC_ARGS="-source 8"
+		einfo "Compiling javadocs"
+		JAVA_SRC_DIR=(
+			"core/src"
+			"uis/src"
+		)
+		JAVA_JAR_FILENAME="ignoreme.jar"
```
Comment (vaukai) [inline net-p2p/biglybt/biglybt-3.6.0.0.ebuild:95]: Use syntax as from [dev-java/apache-rat](https://packages.gentoo.org/packages/dev-java/apache-rat).
`ignoreme.jar` was a hack (See also https://github.com/gentoo/gentoo/pull/36308).

File: net-p2p/biglybt/biglybt-3.6.0.0.ebuild [hunk 2]
```diff
@@ -87,16 +92,9 @@ src_compile() {
...
-		einfo "Compiling javadocs"
-		JAVA_SRC_DIR=(
-			"core/src"
-			"uis/src"
-		)
-		JAVA_JAR_FILENAME="ignoreme.jar"
-		java-pkg-simple_src_compile
-	fi
+	# javadoc is broken for java 17, see https://bugs.gentoo.org/914458
+	JAVADOC_CLASSPATH="${JAVA_GENTOO_CLASSPATH}"
```
Comment (vaukai) [inline net-p2p/biglybt/biglybt-3.6.0.0.ebuild:96]: Is `JAVADOC_CLASSPATH` really needed? Or can do without?

**Review by vaukai — COMMENTED** [review]
The commit message should better read `Modernize javadoc` instead of `Fixed javadoc` since it wasn't broken.

File: net-p2p/biglybt/biglybt-3.6.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	einfo "Compiling module \"uis\""
+	JAVA_JAR_FILENAME="BiglyBT.jar"
+	JAVA_LAUNCHER_FILENAME="${PN}"
+	JAVA_MAIN_CLASS="com.${PN}.ui.Main"
+	JAVA_RESOURCE_DIRS="uis/resources"
+	JAVA_SRC_DIR="uis/src"
+	java-pkg-simple_src_compile
+	rm -r target || die
+
+	# javadoc is broken for java 17, see https://bugs.gentoo.org/914458
```
[the remark this answers is not in the document]
Comment (vaukai) [reply net-p2p/biglybt/biglybt-3.6.0.0.ebuild]: > I thought I might add it as [apache-rat](https://github.com/gentoo/gentoo/blob/823568f4fef6abcbcaef9c7c85d1b5202e2b498f/dev-java/apache-rat/apache-rat-0.15-r1.ebuild#L75) had it.

but that bud has nothing to do with `biglybt`, causes only confusion here ...

**Comment by vaukai** [comment]
> As of right now USE=doc does not work and I don't know why so I need help. Here are the relevant build logs: https://pastebin.com/znrhnh2G
> 
`package org.json.simple does not exist` This is the purpose of putting `JAVADOC_CLASSPATH`.

**Comment by vaukai** [comment]
> Squashed to one commit. I will try to modernize the javadoc now
> 
> EDIT: I do not know how to modernize the JAVADOC and I do not want to work on it in this PR. I think the other changes here are of quality.

[biglybt-3.6.0.0.ebuild.txt](https://github.com/user-attachments/files/15525421/biglybt-3.6.0.0.ebuild.txt)

File: net-p2p/biglybt/biglybt-3.6.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	JAVA_JAR_FILENAME="BiglyBT.jar"
+	JAVA_LAUNCHER_FILENAME="${PN}"
+	JAVA_MAIN_CLASS="com.${PN}.ui.Main"
+	JAVA_RESOURCE_DIRS="uis/resources"
+	JAVA_SRC_DIR="uis/src"
+	java-pkg-simple_src_compile
+
+	if use doc; then
+		einfo "Creating javadoc"
+		JAVADOC_CLASSPATH="${JAVA_GENTOO_CLASSPATH}"
```
Comment (2Kmm) [inline net-p2p/biglybt/biglybt-3.6.0.0.ebuild:87]: `JAVADOC_CLASSPATH` extends the classpath used for the last `java-pkg-simple_src_compile`. So there is no need to set it here at all.

File: net-p2p/biglybt/biglybt-3.6.0.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+		ejavadoc
+	fi
+}
+
+src_install() {
+	java-pkg-simple_src_install
```
Comment (2Kmm) [inline net-p2p/biglybt/biglybt-3.6.0.0.ebuild:93]: `java-pkg-simple_src_install` only installs the jar currently set in `JAVA_JAR_FILENAME`. You have to call `java-pkg_dojar "${PN}-core.jar"` additionally.

Comment (vaukai) [reply net-p2p/biglybt/biglybt-3.6.0.0.ebuild:93]: > `java-pkg-simple_src_install` only installs the jar currently set in `JAVA_JAR_FILENAME`. You have to call `java-pkg_dojar "${PN}-core.jar"` additionally.

No. `biglybt-core.jar` is only a compile dependency.

Comment (2Kmm) [reply net-p2p/biglybt/biglybt-3.6.0.0.ebuild:93]: Ah, thanks for the clarification. :)

File: net-p2p/biglybt/biglybt-3.6.0.0.ebuild (hunk 4 above)
Comment (2Kmm) [reply net-p2p/biglybt/biglybt-3.6.0.0.ebuild:87]: Oh, I apologize. Content of `$JAVADOC_CLASSPATH` is added to `$classpath` and `$classpath` is set in `java-pkg-simple_src_compile` with the help of `$JAVA_GENTOO_CLASSPATH`. What I didn't realized is, that in `java-pkg-simple_src_compile` `$classpath` is a local variable and therefore it is not set, when used outside of this function.

Sorry, you are absolutely right. I should have verified it before posting here.

**Comment by vaukai** [comment]
> @vaukai @2Kmm is this good to go?

i am not a dev, so my opinion is meaningless:
i do not understand the purpose of »`Introduced more ${PN}«
`

**Pull request closed without merging**